src/odemis/driver/technolution.py

```diff
@@ -1431,34 +1432,29 @@ def _setCellDigitalGain(self, cellDigitalGain):
                                  "Digital gain parameters value remain unchanged.")
 
             for column, DigitalGain in enumerate(cellDigitalGain_row):
-                if isinstance(DigitalGain, int):
-                    # Convert all input values to floats.
-                    logging.warning("Input integer values for the digital gain are converted to floats.")
-                    cellDigitalGain = tuple(tuple(float(cellDigitalGain[i][j]) for j in range(0, self.shape[0]))
-                                            for i in range(0, self.shape[0]))
-                    # Call the setter again with all int values converted to floats and return the output
-                    return self._setCellDigitalGain(cellDigitalGain)
```

Comment:
With deleting this we no longer accept integers as an input for the digital gain, for example if the number 1 is entered and somewhere rounded an error could be raised (due to the wrong type being send to the ASM or the VA complaining about containing different types, I don't know which of these was the problem out of my head).
But maybe some things changed and it is now possible to do this though.

p.s. The reason I return at line 1440 with itself is that after updating it to the right format you want to properly perform all checks again.

Reply:
We still accept integers as input (see modification in line 1435). I tested all the testcases against the simulator (so I specifically uploaded the values after each change in the test) and all worked fine also with integers. I didn't find that the VA complains about different types.

Reply:
> p.s. The reason I return at line 1440 with itself is that after updating it to the right format you want to properly perform all checks again.
For readability, instead of returning, I converted all the elif statements to individual if statements. Thus all checks are still performed.

Reply:
A side note: When you run this test, it will not detect whether you actually converted to a float or not. It will just see that 0 is 0.0 and passes. If you want to check the type was converted, you need to explicitly check for the type.
```
# Test int as type for input value
self.MPPC.cellDigitalGain.value = tuple(
        tuple(int(0) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
self.assertEqual(self.MPPC.cellDigitalGain.value,
                 tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
```
You can check it with this line:
```
self.assertEqual(int(0), 0.0)  # will pass
self.assertEqual(type(int(0)), type(0.0))  # will fail
```